test/e2e/storage/utils/utils.go

```diff
@@ -567,8 +572,13 @@ func CheckWriteToPath(f *framework.Framework, pod *v1.Pod, volMode v1.Persistent
 
 	encoded := base64.StdEncoding.EncodeToString(genBinDataFromSeed(len, seed))
 
-	e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | sha256sum", encoded))
-	e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | dd of=%s %s bs=%d count=1", encoded, pathForVolMode, oflag, len))
+	if framework.NodeOSDistroIs("windows") {
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | sha256sum", encoded))
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d > %s", encoded, pathForVolMode))
+	} else {
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | sha256sum", encoded))
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | dd of=%s %s bs=%d count=1", encoded, pathForVolMode, oflag, len))
+	}
```

Comment:
the call:
```
e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("echo %s | base64 -d | sha256sum", encoded))
```

looks the same for both OSes. it can be extracted outside of the branching here.

if not it in this PR, long term it feels better to use only portable methods / commands between both OSes.
(e.g. avoid using `dd` on Linux), to minimize the branching this PR introduces.

Reply:
i have a msys environment installed on Windows and `dd` is present here.
is `dd` missing in agnhost?

Reply:
it's not missing and it's working just fine for the storage test cases in windows, for reference I'm pasting the commands that work:

```
Mar 31 16:57:29.003: INFO: ExecWithOptions {Command:[/bin/sh -c echo wh90iOPpVEQdv9bo+Fjos3mIcsMy2/YkV/PVtAgP+cCrDGzoic30DRSZ+KQO1gznKEhzrQLRYkBZ/YQpeL+EhA== | base64 -d | dd of=/mnt/volume1/file1.txt  bs=64 count=1] Namespace:multivolume-9333 PodName:pod-6a392278-b769-4d02-b8b4-47b965adefe2 ContainerName:write-pod Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false Quiet:false}
Mar 31 16:57:29.003: INFO: >>> kubeConfig: /usr/local/google/home/mauriciopoppe/.kube/config
STEP: Checking if read from the volume1 works properly
Mar 31 16:57:29.484: INFO: ExecWithOptions {Command:[/bin/sh -c dd if=/mnt/volume1/file1.txt  bs=64 count=1 | sha256sum] Namespace:multivolume-9333 PodName:pod-6a392278-b769-4d02-b8b4-47b965adefe2 ContainerName:write-pod Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false Quiet:false}
Mar 31 16:57:29.484: INFO: >>> kubeConfig: /usr/local/google/home/mauriciopoppe/.kube/config
Mar 31 16:57:29.922: INFO: ExecWithOptions {Command:[/bin/sh -c dd if=/mnt/volume1/file1.txt  bs=64 count=1 | sha256sum | grep -Fq 5e95e3618d71d2edeea25e34a2afc2070d4981021929edc2e532cfdc4253b8f7] Namespace:multivolume-9333 PodName:pod-6a392278-b769-4d02-b8b4-47b965adefe2 ContainerName:write-pod Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false Quiet:false}
```

Reply:
thanks,
since i don't know how these windows volumes work, i pinged more folks for review.